test/AST-Quake/vector-ctor-initlist.cpp

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// Test for std::vector initializer_list constructor support
+
+// RUN: cudaq-quake %s | cudaq-opt | FileCheck %s
+
+#include "cudaq.h"
+
+__qpu__ void test() {
+  cudaq::qubit q;
+  std::vector<double> angle{M_PI_2, M_PI_4};
+  ry(angle[0], q);
```

Comment:
Will it handle something that forms the constants via expressions? Do you think it would be worth adding this as a test? Or is it safe to assume that the AST will reduce this for us?
```
std::vector<double> angle{M_PI_2, 22.5*M_PI/180};

Reply:
Yes it will support this as is. I'll update the test to throw an expr like that in there.